torchvision/prototype/transforms/_transform.py

```diff
@@ -36,8 +36,19 @@ def forward(self, *inputs: Any) -> Any:
 
         self._check_inputs(flat_inputs)
 
-        params = self._get_params(flat_inputs)
+        needs_transform_list = self._needs_transform_list(flat_inputs)
+        params = self._get_params(
+            inpt for (inpt, needs_transform) in zip(flat_inputs, needs_transform_list) if needs_transform
+        )
```

Comment:
Note how we're only passing the relevant inputs to `get_params()` now. This is because if we were to pass all the inputs, we'd get a failure down the line in `query_spatial_size()` & Co.

This is because pure tensors are also "queried" there:

https://github.com/pytorch/vision/blob/9b233d41ad71de768a1714eaeb2ebd4f893688e5/torchvision/prototype/transforms/utils.py#L42

which in the case of the **1D** label tensor  would lead to an error:

https://github.com/pytorch/vision/blob/9b233d41ad71de768a1714eaeb2ebd4f893688e5/torchvision/prototype/transforms/functional/_meta.py#L95.

Another way to fix this would be to make `query_spatial_size()` & Co aware of the heuristic somehow. But it's harder because right now, the heuristic relies on `self._transformed_types`. It's probably safer this way for user-defined transforms.